src/components/data/DataTable.tsx

```diff
@@ -58,13 +58,13 @@ const DataTableNext = function DataTable({
   title,
   selectedRow,
   loading = false,
-  renderItem = (row, field) => /** @type {Children} */ (row[field]),
+  renderItem = (r: Row, field: keyof Row) => r[field] as ComponentChildren,
```

Comment:
TS might be able to infer the argument types here, I'm not sure. Either way, it doesn't hurt to specify them.